pkg/osquery/tables/filevault/filevault.go

```diff
@@ -0,0 +1,56 @@
+// +build darwin
+
+package filevault
+
+import (
+	"bytes"
+	"context"
+	"os/exec"
+	"strings"
+
+	"github.com/go-kit/kit/log"
+	"github.com/kolide/osquery-go"
+	"github.com/kolide/osquery-go/plugin/table"
+	"github.com/pkg/errors"
+)
+
+const fdesetupPath = "/usr/bin/fdesetup"
+
+type Table struct {
+	client *osquery.ExtensionManagerClient
+	logger log.Logger
+}
+
+func TablePlugin(client *osquery.ExtensionManagerClient, logger log.Logger) *table.Plugin {
+	columns := []table.ColumnDefinition{
+		table.TextColumn("status"),
+	}
+
+	t := &Table{
+		client: client,
+		logger: logger,
+	}
+
+	return table.NewPlugin("kolide_filevault", columns, t.generate)
+}
+
+func (t *Table) generate(ctx context.Context, queryContext table.QueryContext) ([]map[string]string, error) {
+	var results []map[string]string
+
+	// Read the system's fdesetup configuration
+	var stdout bytes.Buffer
+	stdout.Reset()
+	cmd := exec.CommandContext(ctx, fdesetupPath, "status")
```

Comment:
This should get a timeout. This is done through the `ctx` See https://github.com/kolide/launcher/blob/faa8c090d04ac7a904ede37d34127fa5de906d5f/pkg/osquery/table/mdfind_darwin.go#L15-L16